### PR TITLE
Include capability to run official builds

### PR DESCRIPTION
**Pull request by mbacchi**
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -261,12 +279,42 @@ pipeline {
...
+                        stage("dist-release") {
+                            when {
+                                expression { env.LABEL_SUFFIX == 'release' }
+                            }
+                            steps {
+                                sh "npm run create_dist -- ${BUILD_TYPE} --channel=${CHANNEL} --debug_build=false --official_build=true"
+                            }
+                        }
+                        // TODO: add upload step here
+                        stage("github_upload") {
```
Comment (mihaiplesa) [inline Jenkinsfile]: `github-upload` instead of `github_upload`

File: Jenkinsfile [hunk 2]
```diff
@@ -261,12 +279,42 @@ pipeline {
...
+                                expression { env.LABEL_SUFFIX == 'release' }
+                            }
+                            steps {
+                                withCredentials([[
+                                    $class: 'AmazonWebServicesCredentialsBinding',
+                                    credentialsId: 'brave-browser-binaries-upload',
+                                    accessKeyVariable: 'BRAVE_S3_ACCESS_KEY',
+                                    secretKeyVariable: 'BRAVE_S3_SECRET_KEY'
+                                ]]) {
+                                    sh 'PATH=${PATH}:/usr/local/bin/ npm run upload'
```
Comment (mihaiplesa) [inline Jenkinsfile]: we can move PATH setting into a new `environment` section per platform

File: Jenkinsfile [hunk 3]
```diff
@@ -277,11 +325,29 @@ pipeline {
...
+                                    )
+                                }
+                            }
+                        }
                     }
                 }
                 stage("windows-x64") {
                     agent { label "windows-${LABEL_SUFFIX}" }
                     environment {
+                        BRAVE_S3_BUCKET = "brave-brave-binaries"
```
Comment (mihaiplesa) [inline Jenkinsfile]: defined at top-level, not needed here anymore

File: Jenkinsfile [hunk 4]
```diff
@@ -277,11 +325,29 @@ pipeline {
                                 }
                             }
                         }
+                        stage("archive-release") {
```
Comment (mihaiplesa) [inline Jenkinsfile]: `archive` can be one single step as discussed going into `${OUT_DIR}/*.dmg` and `${OUT_DIR}/*.pkg`

File: Jenkinsfile [hunk 5]
```diff
@@ -402,6 +468,21 @@ pipeline {
                                 powershell "npm run create_dist -- ${BUILD_TYPE} --channel=${CHANNEL} --build_omaha --tag_ap=x64-${CHANNEL} --target_arch=x64 --debug_build=false --official_build=true"
                             }
                         }
+                        stage("github_upload") {
```
Comment (mihaiplesa) [inline Jenkinsfile]: same as above, please use `-` instead of `_`

File: Jenkinsfile [hunk 6]
```diff
@@ -431,6 +512,179 @@ pipeline {
...
                     }
                 }
+                stage("windows-ia32") {
+                    when {
+                        beforeAgent true
+                        expression { env.LABEL_SUFFIX == 'release' }
+                    }
+                    agent { label "windows-${LABEL_SUFFIX}" }
+                    environment {
+                        BRAVE_S3_BUCKET = "brave-brave-binaries"
```
Comment (mihaiplesa) [inline Jenkinsfile]: defined at top-level, not needed here anymore

File: Jenkinsfile [hunk 7]
```diff
@@ -431,6 +512,179 @@ pipeline {
                         }
                     }
                 }
+                stage("windows-ia32") {
+                    when {
```
Comment (mihaiplesa) [inline Jenkinsfile]: nice one here!

File: Jenkinsfile [hunk 8]
```diff
@@ -431,6 +512,179 @@ pipeline {
...
+                                ]]) {
+                                    powershell 'npm run upload -- --target_arch=ia32'
+                                }
+                            }
+                        }
+                        stage("archive") {
+                            steps {
+                                // commented because it takes much longer to copy to Jenkins thant to S3
+                                // archiveArtifacts artifacts: "${OUT_DIR}/BraveBrowser*.exe", fingerprint: true
+                                s3Upload(acl: "Private", bucket: "${BRAVE_ARTIFACTS_BUCKET}", includePathPattern: "brave_installer*.exe",
```
Comment (mihaiplesa) [inline Jenkinsfile]: might have to put specific installer file name here for 32-bit

Comment (mbacchi) [reply Jenkinsfile]: Both win64 and win32 installer files are called `brave-installer.exe`, which was incorrect in the current pattern match. This meant that the 64 bit installer was not being archived to the s3 bucket for artifacts in previous builds.

Comment (mihaiplesa) [reply Jenkinsfile]: the 64-bit one that gets uploaded is `brave_installer_72_0_62_0.exe` so we need that wildcard

File: Jenkinsfile [hunk 9]
```diff
@@ -13,6 +13,8 @@ pipeline {
         booleanParam(name: 'CLEAN_WORKSPACE', defaultValue: false)
     }
     environment {
+        BRAVE_S3_BUCKET = "brave-brave-binaries"
```
Comment (mihaiplesa) [inline Jenkinsfile]: these 2 should go at the bottom of the section

**Pull request merged**